### PR TITLE
Commit inital sdk logic w/ Readme

### PR DESCRIPTION
**Pull request by DyrellC**
This PR represents the start point for the Rust SDK for `Alvarium`. A README is included to provide basic usage instructions and relay concepts regarding the abstraction of `Annotators` and `Publishers`. 

There are also 30 unit/regression tests present to ensure future changes can be checked against previous iterations to ensure breaking changes aren't introduced accidentally.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+[dependencies]
+tokio = { version = "1.32.0", features = ["rt", "macros"] }
+chrono = "0.4.22"
+lazy_static = "1.4.0"
+
+md5-rs = "0.1.5"
+hex = "0.4.3"
+base64 = "0.13.0"
+
+alvarium-annotator = { git = "https://github.com/DyrellC/alvarium-annotator" }
```
Comment (tsconn23) [inline Cargo.toml]: @DyrellC Does this reference need to be updated now that the `alvarium-annotator` repo is populated?

File: README.md [hunk 2]
```diff
@@ -1 +1,244 @@
...
+
+Examples of stream provider configurations can be found [here](resources/mqtt_stream_config.json) 
+for mqtt and [here](resources/demia_streams_config.json) for a Demia (powered by IOTA) Streams provider.
+
+
+To include the rust sdk in your project insert the following into your Cargo.toml file
+
+```
+[dependencies]
+alvarium-annotator = { git = "https://github.com/DyrellC/AlvariumAnnotator" }
```
Comment (tsconn23) [inline README.md]: @DyrellC Looks like another reference to be updated

File: README.md [hunk 3]
```diff
@@ -1 +1,244 @@
...
+alvarium-rust-sdk = { git = "https://github.com/DyrellC/alvarium-rust-sdk" }
```
Comment (tsconn23) [inline README.md]: @DyrellC Here as well

File: README.md [hunk 4]
```diff
@@ -1 +1,244 @@
...
+                Err(alvarium_rust_sdk::errors::Error::NoHostName.into())
+            }
+        }
+    }
+}
+```
+
+#### Custom Stream Providers
+The base SDK includes an Mqtt and Demia Streams provider, but new stream providers can be created 
+using the alvarium-annotator [Publisher](https://github.com/DyrellC/alvarium-annotator/blob/main/src/providers.rs#L124)
```
Comment (tsconn23) [inline README.md]: @DyrellC I guess just do a scan for all DyrellC github references and replace them

File: Cargo.toml (hunk 1 above)
Comment (DyrellC) [reply Cargo.toml]: Yes all of these will be updated to use the appropriate repo's now that https://github.com/project-alvarium/alvarium-annotator/pull/1 has been updated

**Review by tsconn23 — APPROVED** [review]

**Pull request merged**